### ISSUE TITLE
`explicit-length-check` rule should not be triggered when length is part of a numeric expression

### ISSUE THREAD

**Issue by broofa** [sindresorhus/eslint-plugin-unicorn#1933]
The `explicit-length-check` rule is getting triggered in the following context where the length is actually being used for it's numeric value rather than as a boolean.  Telling the user to convert it to a logical comparison expression is not appropriate.

<img width="708" alt="CleanShot 2022-10-21 at 15 41 12@2x" src="https://user-images.githubusercontent.com/164050/197298880-4390ed0a-50c0-4b0e-9524-3b455d4e2b19.png">

`explicit-length-check`

```js
  const x = [1,2].length || 1;
```

<img width="735" alt="CleanShot 2022-10-21 at 15 44 20@2x" src="https://user-images.githubusercontent.com/164050/197299102-9c135f5a-084f-4087-a6b1-6aba79d53008.png">

**Comment by yoyo837** [comment]
Same here.
